bin/go-rpm-integration

```diff
@@ -155,45 +154,49 @@ while [ $# -gt 0 ] ; do
 done
 
 installfile() {
-  goipath="${1}"
-  file="${2}"
-  file="${file#./}"
-  [[ -d "${file}" && ! -L "${file}" ]] && srcdir="${file}" || srcdir=$(dirname "${file}")
-  destdir="${prefix}${gopath}/src/${goipath}/${srcdir}"
-  destdir="${destdir%/.}"
-  dir="${destdir}"
-  dirs=()
-  while [[ ! -e "${dir}" ]] ; do
-    dirs=("$dir" "${dirs[@]}")
-    dir=$(dirname "${dir}")
-  done
-  for dir in "${dirs[@]}" ; do
-    install -m 0755 -vd "${dir}"
-    if $(echo "${dir}" | grep -q "^${prefix}${gopath}/src/${goipath}") ; then
-      touch -r             ".${dir#${prefix}${gopath}/src/${goipath}}" "${dir}"
-    fi
-    echo "%dir \"${dir#${prefix}}\"" >> ${filelist}
-  done
-  if [[ -L "$file" ]] ; then
-    ln -s $(readlink "${file}") "${destdir}/$(basename ${file})"
-    touch -h      -r "${file}"  "${destdir}/$(basename ${file})"
+goipath="${1}"
+file="${2}"
+for proot in "$(realpath -sm ${PWD})" \
+             "${gopath}/src/${goipath}" \
+             "${prefix}${gopath}/src/${goipath}"; do
+  [[ ${file}/ == ${proot}/* ]] && file=".${file#${proot}}"
+done
+file="$(realpath -sm ${file})"
+workdir="$(realpath -sm .)"
+if [[ ${file}/ == ${workdir}/* ]] ; then
+  dest="${prefix}${gopath}/src/${goipath}${file#${workdir}}"
+else
+  dest="${prefix}${file}"
+fi
+destdir="$(dirname ${dest})"
+if [[ ! -d "${destdir}" ]] ; then
+  installfile "${goipath}" "${destdir#${prefix}}"
+fi
+if [[ (! -e $file) || (-d "${file}" && ! -L "${file}") ]] ; then
+  install -m 0755 -vd "${dest}"
+  if [[ -d "${file}" && ! -L "${file}" ]] ; then
+    find "${file}" -iname '*.md' -type f -maxdepth 1 | while read docfile ; do
```

Comment:
Are you sure the *.md files belongs under `%{goipath}` directory? Unless they are needed by the code itself, they should live under %doc.

Reply:
Putting them in %doc would work if so many projects didn't put md files deep inside their project tree (and call them all README.md or some such generic name that collides)

This way the md files are still marked as %doc but keep their original place in the project tree